R/utils.R

```diff
@@ -1057,3 +1057,30 @@ countAndPrint <- function(x) {
   if (inherits(object, "OnDiskMSnExp")) msLevel(object)[1]
   else msLevel(object[[1]])
 }
+
+#' @title Define the type of mzR backend to use based on the file name
+#'
+#' @description Simple helper to define the mzR backend that should/can be used
+#'     to read the file.
+#'
+#' @param x \code{character(1)} representing the file name.
+#'
+#' @return A \code{character(1)} with the name of the backend (either
+#'     \code{"netCDF"}, \code{"Ramp"} or \code{"pwiz"}.
+#' 
+#' @author Johannes Rainer
+#'
+#' @noRd
+mzRBackend <- function(x) {
```

Comment:
AFAIK `mzR::openMSfile` supports just one filename and just one single backend. That's why something like `mzRBackend(c("foo.mzML", "foo.xml", "foo.nc"))` should fail (and not return `c("pwiz", "netCDF")`). Or it should be completely vectorized and return `c("pwiz", "pwiz", "netCDF")`. But IMHO vectorization is overkill here (because `mzR::openMSfile` expects one single filename and backend).